portfolio/src/main/java/com/google/sps/data/CalendarGroup.java

```diff
@@ -0,0 +1,132 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/** This class models a group of calendar events. */
+public class CalendarGroup {
+  public List<CalendarEvent> events;
+  public List<TimeRange> allFreeTimeRanges;
+
+  // These two variables represent the start and end of
+  // possible schedule time. When these times are passed in
+  // to instantiate a CalendarGroup class, these times are
+  // already adjusted based on user's working hour times
+  // and the current timestamp of when the user starts
+  // the scheduling process.
+  public final Instant startTime;
+  public final Instant endTime;
+
+  // Comparator for sorting events by start time
+  private static final Comparator<CalendarEvent> sortByEventStartTimeAscending =
+      Comparator.comparing(CalendarEvent::getStartTime);
+
+  // Comparator for sorting free time ranges by start time
+  private static final Comparator<TimeRange> sortByTimeRangeStartTimeAscending =
```

Comment:
Maybe we can move this into the TimeRange class.

Reply:
Updated.